index.js

```diff
@@ -119,7 +103,46 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^([^]+)\/([^]+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (Nullable String, String, Number)
+  //                      -> { namespace :: Nullable String
+  //                         , name :: String
+  //                         , version :: Number }
+  function TypeIdentifier(namespace, name, version) {
+    return {namespace: namespace, name: name, version: version};
+  }
+
+  //. ### Usage
+  //.
+  //. `const {parse, type} = require('sanctuary-type-identifiers')`
+  //.
+  //. ```javascript
+  //. > parse(type([1, 2, 3]));
+  //. {namespace: null, name: 'Array', version: 0}
+  //. ```
+  //.
+  //. ### API
+  //.
+  //# type :: Any -> String
+  //.
+  //. Takes any value and produces a String which represents its type. If the
+  //. value conforms to the [specification][3], the custom type identifier is
+  //. returned.
+  //.
+  //. ```javascript
+  //. > type(null);
+  //. 'Null'
+  //.
+  //. > type(true);
+  //. 'Boolean'
+  //.
+  //. > function Identity(x) { this.value = x; }
+  //. > Identity['@@type'] = 'my-package/Identity@1';
+  //. > type(new Identity(0));
+  //. 'my-package/Identity@1'
```

Comment:
The use of `new` in the usage example bothers me. I'd prefer to make the set-up noisier to avoid it:

```javascript
  //. > function Identity(x) {
  //. .   if (!(this instanceof Identity)) return new Identity(x);
  //. .   this.value = x;
  //. . }
  //. . Identity['@@type'] = 'my-package/Identity@1';
  //.
  //. > type(Identity(42))
  //. 'my-package/Identity@1'
```